### PR TITLE
Add a separate setting for enabling DNS names in pod locality

### LINKED ISSUE #1230: Setting DNS configuration on pods independently of using DNS in the cluster file
### What would you like to be added/changed?

When we enable DNS names in the cluster file, we do a rolling bounce to recreate the pods in order to add the necessary DNS information. This slows down the process, and slows down our testing. It should be harmless to add all of the DNS metadata before someone tries to convert the cluster to use DNS names, and that may be desireable in the transition path, because actually using the DNS names requires an FDB upgrade and may require additional testing in the infrastructure. I think we should have an option to enable this DNS configuration independently of using DNS in the cluster file.

As an alternative, if we can find a way to avoid this configuration, such as having the DNS locality field be built off of static configuration and existing dynamic fields, that would help serve the same purpose.

### PR DESCRIPTION
**Pull request by brownleej**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

This allows us to configure the pod spec aspects of the new DNS name support long before updating the cluster file itself, which should provide a smoother transition and easier testing of the conversion process.

Fixes #1230 

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

## Discussion

*Are there any design details that you would like to discuss further?*

No.

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

Manual testing.

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

No.

## Documentation

*Did you update relevant documentation within this repository?*

Yes.

*If this change is adding new functionality, do we need to describe it in our user manual?*

No, I don't think so. It's fairly minor.

## Follow-up

*Does this introduce new defaults that we should re-evaluate in the future?*

We may want to make this new setting default to true in the next major release.

### CODE REVIEW

**Review by johscheuer — APPROVED** [review]

**Review by johscheuer — COMMENTED** [review]
I wonder if those failures are a race condition want we should add an additional "kubectl wait"  to ensure the CRD is properly updated.

**Review by ltsampros — APPROVED** [review]
LGTM

**Review by AnnigeriShambu — APPROVED** [review]

**Review by johscheuer — COMMENTED** [review]
Do you think you have some time to rebase those changes?

**Review by johscheuer — COMMENTED** [review]
@brownleej what do you think are the next steps for this PR?

**Comment by brownleej** [comment]
I think this just got lost in the noise when we shifted our focus onto stabilization last fall. I'll get this PR into a mergeable state.

**Comment by brownleej** [comment]
I've opened a separate PR to try to fix the integration checks.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -95,6 +95,7 @@ jobs:
         # Ensure we can upgrade the CRD with the current changes
         make install
         kubectl wait --for condition="established" crd --all
+        sleep 60
```
Comment (johscheuer) [inline .github/workflows/pull_request.yml:98]: Can wee make a comment here that this is a workaround and we should investigate how to solve this without a sleep?

File: api/v1beta2/foundationdbcluster_types.go [hunk 2]
```diff
@@ -1718,6 +1718,12 @@ type RoutingConfig struct {
 	// latest stable version of FoundationDB.
 	UseDNSInClusterFile *bool `json:"useDNSInClusterFile,omitempty"`
 
+	// DefineDNSLocalityFields determines whether to define pod DNS names on pod
```
Comment (johscheuer) [inline api/v1beta2/foundationdbcluster_types.go:1721]:
```suggestion
	// DefineDNSLocalityFields determines whether to define Pod DNS names on Pod
```

File: api/v1beta2/foundationdbcluster_types.go [hunk 3]
```diff
@@ -1997,6 +2003,12 @@ func (cluster *FoundationDBCluster) UseDNSInClusterFile() bool {
 	return pointer.BoolDeref(cluster.Spec.Routing.UseDNSInClusterFile, false)
 }
 
+// DefineDNSLocalityFields determines whether we need to put DNS entries in the
+// pod spec and process locality.
```
Comment (johscheuer) [inline api/v1beta2/foundationdbcluster_types.go:2007]:
```suggestion
// Pod spec and process locality.
```

**Review by johscheuer — COMMENTED** [review]
Changes LGTM.

File: api/v1beta2/foundationdbcluster_types.go (hunk 2 above)
Comment (brownleej) [reply api/v1beta2/foundationdbcluster_types.go:1721]: We may need a breakout session on whether "pod" is a proper noun for style purposes. 😁

Comment (johscheuer) [reply api/v1beta2/foundationdbcluster_types.go:1721]: Sounds reasonable, even the official docs are inconsistent 🙄 https://kubernetes.io/docs/concepts/workloads/pods/

**Comment by johscheuer** [comment]
> ### Result of fdb-kubernetes-operator-pr on Linux CentOS 7
> * Commit ID: [655df4c](https://github.com/FoundationDB/fdb-kubernetes-operator/commit/655df4cf564dcae16ccbc9420b78eb127b75620b)
> * Duration 3:12:30
> * Result: ❌ **FAILED**
> * Error: `Error while executing command: if $fail_test; then exit 1; fi. Reason: exit status 1`
> * [Build Log](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator/fdb-kubernetes-operator-pr/4747ffc2-8bd9-40dd-b425-23618a582bf0.gz) terminal output (available for 30 days)
> * [Build Workspace](https://d1e0l78xbkh3xa.cloudfront.net/fdb-kubernetes-operator-pr/4747ffc2-8bd9-40dd-b425-23618a582bf0/workspace.zip) zip file of the working directory (available for 30 days)

All tests passed except for one test, that failed with:

```text
ERROR: This exclude may cause the total free space in the cluster to drop below 10%.\nType `exclude FORCE <ADDRESS...>' to exclude without checking free space.\n
```

This is a known issue that should be fixed soon.

**Comment by johscheuer** [comment]
Thanks!

**Pull request merged**